src/uproot/containers.py

```diff
@@ -406,35 +408,106 @@ def typename(self):
         else:
             return self._typename
 
+    # def descent(self, context):
+
     def awkward_form(self, file, context):
+        keys = context.keys()
         awkward = uproot.extras.awkward()
-        return awkward.forms.ListOffsetForm(
-            context["index_format"],
-            awkward.forms.NumpyForm((), 1, "B", parameters={"__array__": "char"}),
-            parameters={
-                "__array__": "string",
-                "uproot": {
-                    "as": "string",
-                    "header": self._header,
-                    "length_bytes": self._length_bytes,
+        if "forth" in keys:
+            forth_obj = context["forth"]
+            forth_obj.forth_code[id(self)] = None
+            key = forth_obj.get_last_key()
+            forth_obj.init_keys(self, key + 1, key + 2)
+            forth_obj.register_pre(self)
+            forth_obj.register_post(self)
+            return awkward.forms.ListOffsetForm(
+                context["index_format"],
+                awkward.forms.NumpyForm(
+                    (),
+                    1,
+                    "B",
+                    parameters={"__array__": "char"},
+                    form_key=f"node{key + 2}",
+                ),
+                parameters={
+                    "__array__": "string",
+                    "uproot": {
+                        "as": "string",
+                        "header": self._header,
+                        "length_bytes": self._length_bytes,
+                    },
                 },
-            },
-        )
+                form_key=f"node{key + 1}",
+            )
+        else:
+            return awkward.forms.ListOffsetForm(
+                context["index_format"],
+                awkward.forms.NumpyForm((), 1, "B", parameters={"__array__": "char"}),
+                parameters={
+                    "__array__": "string",
+                    "uproot": {
+                        "as": "string",
+                        "header": self._header,
+                        "length_bytes": self._length_bytes,
+                    },
+                },
+            )
 
     def read(self, chunk, cursor, context, file, selffile, parent, header=True):
+        forth = False
+        forth_obj = None
+        jump = True
+        fcode = []
+        fheader = []
+        finit = []
+        if "forth" in context.keys():
+            awkward = uproot.extras.awkward()
+            forth = True
+            forth_obj = context["forth"]
+            isinstance(
+                forth_obj.aform,
+                (awkward.layout.NumpyArray,),
+            )
```

Comment:
This code doesn't do anything.

Also, if `aform` is really a Form, it can never have type `NumpyArray`.